### PR TITLE
Joe/feat/navigation buttons

### PR DESCRIPTION
**Pull request by JoeDodgson**
# NEW CHANGES

* Privacy policy:
  - Added privacy policy checkbox on login and signup page with a link to privacy policy HTML route.
  - Created a privacy policy pug file to display a simply-formatted privacy notice
  - Created a privacy policy HTML which renders privacyNotice pug 
  - Added a 'warning' message to display when a user tries to login / sign up without agreeing to privacy policy. Has a display: none style so it does not display by default.
  - Somebody will need to write the JS to restrict a user from logging in when they haven't checked the privacy policy box. Also, when they try, remove the display: none style from the 'warning' message

* Styling:
  - Removed jumbotron from login and signup pages on small screens
  - Reduced border radius on jumbotron to 5px and changed box shadow to green
  - Moved search and favourites buttons inside the jumbotron. Changed focus colour to white and button focus shadow to white
  - Changed justification of exercises on search results page
  - Moved all styling from pug into CSS
  - Removed width values from form inputs 

* Welcome message:
  - Moved welcome message to inside top of div
  - Still need to work out how to pass data into res.redirect from the login and signup routes. Tried to use flash npm module, but that's not working...

# Summary of changes

* Changes across all pages:
  - Amended logout link on each page to include a href for api/logout, to increase the size and to underline when hovered over
  - Added a 'favourites' and 'search' nav buttons across each page (except for the pages themselves), along with hover styling

* Login page:
  - Added a signup button with link to signup page
  - Removed 'sign in with Facebook' message and the green horizontal break line
  - Added 'login' heading

* Signup page:
  - Added space character between 'login' and 'here'

* Favourite exercises page:
  - Added a welcome message, including the user's name (passed in user's name in the html renderer)
  - Linked each individual exercise to its exercise details page
  - Added hover styling for each individual exercise so user know it's clickable

* Exercise details page:
  - Removed second heading- used the exercise name as the heading
  - Embedded a video link. This is optional in the template engine so if no video is provided, the HTML element won't appear

* Search results page:
  - Changed the heading to 'Search results'
  - Centred the exercise names
  - Linked exercise to their detail pages and added hover styling
  - Added styling to stars so that if the user has favourited the star is filled and green. If not, the star is white and not filled
  - Added hover styling to filled star - becomes white on hover

* HTML routes:
  - Added dummy data for favourite (boolean depending on if the user has favourited that exercise)
  - Added dummy data for exercise video URL

# Testing
* Read code changes
* Connect to JAWSDB from local workbench
* Test deployment
* Check that all pages render ok and are responsive at different screen sizes (Note: the only page which is not navigable via clicks is /search/results)
* Check that all links work
* Give feedback on styling

### CODE REVIEW

File: public/stylesheets/less/style.less [hunk 1]
```diff
@@ -53,11 +55,27 @@
   text-decoration: none;
 }
 
-// Mixin for main gray centre div
+// Mixin for main gray center div
 .center-div {
   background: @tertiaryColor;
   border-radius: 20px;
```
[the remark this answers is not in the document]
Comment (Goldenaries658) [reply public/stylesheets/less/style.less:61]: Reduce border-radius, to be more clear

File: public/stylesheets/less/style.less [hunk 2]
```diff
@@ -2,7 +2,7 @@
 @primaryColor: #6cad2e;
 @secondaryColor: #444;
 @tertiaryColor: #66666666;
-@red: #b03060;
+@red: rgb(189, 45, 45);
```
[the remark this answers is not in the document]
Comment (JoeDodgson) [reply public/stylesheets/less/style.less]: Agreed. Code amended

File: public/stylesheets/less/style.less (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoeDodgson) [reply public/stylesheets/less/style.less]: Agreed. Code amended

File: public/stylesheets/less/style.less [hunk 3]
```diff
@@ -53,11 +55,27 @@
   text-decoration: none;
 }
 
-// Mixin for main gray centre div
+// Mixin for main gray center div
 .center-div {
```
[the remark this answers is not in the document]
Comment (JoeDodgson) [reply public/stylesheets/less/style.less:52]: It still exists

**Pull request merged**